libp2p/Host.cpp

```diff
@@ -120,6 +120,15 @@ Host::~Host()
 void Host::start()
 {
 	startWorking();
+	while (isWorking() && !haveNetwork())
+		this_thread::sleep_for(chrono::milliseconds(10));
```

Comment:
isWorking will be set to false if/when startWorking fails due to an exception; if the exception doesn't occur then haveNetwork will return true.